tests/rptest/tests/replication_factor_change_test.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+
+import time
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.rpk import RpkTool, RpkException
+from rptest.services.admin import Admin
+from rptest.services.redpanda import CHANGE_REPLICATION_FACTOR_ALLOW_LIST
+
+
+class ReplicationFactorChangeTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=3, replication_factor=3), )
+
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_leader_balancer": False,
+            "partition_autobalancing_mode": "off",
+        }
+
+        super(ReplicationFactorChangeTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=4,
+                             extra_rp_conf=extra_rp_conf)
+
+        self._rpk = RpkTool(self.redpanda)
+        self.admin = Admin(self.redpanda)
+
+        self.rf_property = "replication.factor"
+        self.topic_name = self.topics[0].name
+
+    def check_rf(self):
+        for partition in self._rpk.describe_topic(self.topic_name):
+            if len(partition.replicas) != self.replication_factor:
+                return False
+        return True
+
+    @cluster(num_nodes=4)
+    def simple_test(self):
+        self.replication_factor = 4
+        self._rpk.alter_topic_config(self.topic_name, self.rf_property,
+                                     self.replication_factor)
+        wait_until(self.check_rf,
+                   timeout_sec=60,
+                   backoff_sec=2,
+                   err_msg="Can not change replication factor")
+
+        def wait_rec():
+            return len(self.admin.list_reconfigurations()) == 0
+
+        wait_until(wait_rec,
+                   timeout_sec=60,
+                   backoff_sec=2,
+                   err_msg="Can not wait end of reconfiguration")
+
+        self.replication_factor = 2
+        self._rpk.alter_topic_config(self.topic_name, self.rf_property,
+                                     self.replication_factor)
+        wait_until(self.check_rf,
+                   timeout_sec=60,
+                   backoff_sec=2,
+                   err_msg="Can not change replication factor")
+
+    @cluster(num_nodes=4, log_allow_list=CHANGE_REPLICATION_FACTOR_ALLOW_LIST)
+    def check_error_test(self):
+        self.replication_factor = -1
+        try:
+            self._rpk.alter_topic_config(self.topic_name, self.rf_property,
+                                         self.replication_factor)
+            assert False, "We can not set replication factor <= 0"
```

Comment:
Can these assertions also check the message from rpk?  We should be validating that it's something well-formed and not e.g. an unknown_server_error